src/index.ts

```diff
@@ -0,0 +1,30 @@
+/*--------------------------------------------------------------------------
+
+@sinclair/typebox
+
+The MIT License (MIT)
+
+Copyright (c) 2022 Haydn Paterson (sinclair) <haydn.developer@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+
+---------------------------------------------------------------------------*/
+
+export * from './typebox';
+export * as Guards from './guards/index';
```

Comment:
Ah, let's omit the `export` for the guards here.

Note: the `compiler`, `guards` and `value` modules are new and intended to be imported with the following.

```typescript
import { TypeCompiler } from '@sinclair/typebox/compiler'
import { TypeGuard } from '@sinclar/typebox/guards'
import { Value }  from '@sinclar/typebox/value' 
```
The reason for this primarily stems from having the ability to tree shaking where users can optionally pull this additional infrastructure if they need it. This mostly relates to the compiler module (where users would typically be using Ajv, and thus should be optionally bundled).

The general rule moving forward is that if there is any runtime functionality (or anything that results in additional code that isn't erased through compilation), then it needs to be treated optionally.

Would be open to additional approaches if you have any ideas tho.